### PR TITLE
ci: re-enable windows builds

### PR DESCRIPTION
**Pull request by brad-mysten**

### CODE REVIEW

**Comment by huitseeker** [comment]
@brad-mysten is this still useful?

**Comment by brad-mysten** [comment]
> @brad-mysten is this still useful?

Yeah, I will move it out of draft state once we have the self-hosted windows builder up and running.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -60,14 +60,27 @@ jobs:
...
+          - [self-hosted, self-hosted-ubuntu]
+          - [self-hosted, windows]
       fail-fast: false
     steps:
+      # Set the path for windows builds: the runner needs to be able to find
+      # llvm binaries as well as the "bash" binary included with msys64.
+      - name: Adjust path for windows builds
+        if: "startsWith(runner.os, 'windows')"
+        run: |
+          echo "C:\Program Files\llvm\bin;C:\msys64\usr\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
```
Comment (brad-mysten) [inline .github/workflows/rust.yml:77]: I can actually add these to the system path for the build user, which is probably better.

Comment (bmwill) [reply .github/workflows/rust.yml:77]: Yeah anything we can do on the machine itself to make the jobs as identical as possible would be awesome.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -60,14 +60,27 @@ jobs:
...
       - uses: actions/checkout@v3
       - uses: actions-rs/toolchain@v1
+      # The install script for nextest doesn't work on windows (very linux/bash
+      # specific) so the windows builders already have the binaries installed.
       - uses: taiki-e/install-action@nextest
+        if: "startsWith(runner.os, 'linux')"
```
Comment (bmwill) [inline .github/workflows/rust.yml:83]: the docs for this action state that it should work on windows. is this another case of deps just not being installed on the runner?

Comment (brad-mysten) [reply .github/workflows/rust.yml:77]: I tried, ends up not working well on even windows bash :shrug:

File: .github/workflows/rust.yml (hunk 1 above)
Comment (brad-mysten) [reply .github/workflows/rust.yml]: Added to system path and removed

**Review by tharbert — APPROVED** [review]

**Pull request merged**